src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -185,6 +185,23 @@ def next_slice_id(self) -> str:
         self.slice_id += 1
         return result
 
+    def slice_info_for_literal(self, length, prefix="") -> RawSliceInfo:
+        """Returns a RawSliceInfo for a literal.
+
+        In the alternate template, literals are replaced with a uniquely
+        numbered, easily-to-parse literal. JinjaTracer uses this output as
+        a "breadcrumb trail" to deduce the execution path through the template.
+
+        This is important even if the original literal (i.e. in the raw SQL
+        file) was empty, as is the case when Jinja whitespace control is used
+        (e.g. "{%- endif -%}"), because fewer breadcrumbs means JinjaTracer has
+        to *guess* the path, in which case it assumes simple, straight-line
+        execution, which can easily be wrong with loops and conditionals.
+        """
+        unique_alternate_id = self.next_slice_id()
+        alternate_code = f"\0{prefix}{unique_alternate_id}_{length}"
+        return RawSliceInfo(unique_alternate_id, alternate_code, [])
```

Comment:
@tunetheweb: Rather than just commenting the two areas of added code, I refactored it into a new function that is also useful one other place in the code. This is cleaner overall, and avoids needing the same comment in two places.